.codegen/impl.java.tmpl

```diff
@@ -26,18 +28,36 @@ class {{.PascalName}}Impl implements {{.PascalName}}Service {
     String path = {{if .PathParts -}}
       String.format("{{range  .PathParts}}{{.Prefix}}{{if or .Field .IsAccountId}}%s{{end}}{{ end }}"{{ range .PathParts }}{{if .Field}}, request.get{{.Field.PascalName}}(){{ else if .IsAccountId }}, apiClient.configuredAccountID(){{end}}{{ end }})
     {{- else}}"{{.Path}}"{{end}};
-    {{if .Response -}}
-      {{- if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class);
-      {{- else if .Response.MapValue -}} return apiClient.getStringMap(path, request);
-      {{- else -}} return apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-                        {{- else if .Response.MapValue }}Map
-                        {{- else}}{{template "type" .Response}}
-                        {{- end -}}{{else}}, Void{{end}}.class);{{end}}
-    {{- else -}}apiClient.{{.Verb}}(path{{if .Request}}, request{{end}}{{if .Response -}}, {{if .Response.ArrayValue }}Collection
-      {{- else if .Response.MapValue }}Map
-      {{- else}}{{template "type" .Response}}
-      {{- end -}}{{else}}, Void{{end}}.class);
-    {{end}}
+    {{ template "headers" . -}}
+    {{ if not .Response -}}
+    {{ template "api-call" . }}
+    {{- else if .Response.ArrayValue -}} return apiClient.getCollection(path, null, {{template "type" .Response.ArrayValue}}.class, headers);
+    {{- else if .Response.MapValue -}} return apiClient.getStringMap(path, {{ template "request-param" .}}, headers);
+    {{- else if .IsResponseByteStream -}}
+    InputStream response = {{ template "api-call" . }}
```

Comment:
why not returning `InputStream` as a result type? it makes sense to return a wrapping response type only if there's more than one field in it, otherwise it's confusing.